common/src/test/java/io/druid/collections/StupidPoolTest.java

```diff
@@ -72,10 +70,24 @@ public void testExceptionInResourceHolderGet() throws IOException
     resourceHolderObj.get();
   }
 
-  @Test
-  public void testFinalizeInResourceHolder()
+  @Test(timeout = 60_000)
+  public void testResourceHandlerClearedByJVM()
+  {
+    if (System.getProperty("java.version").startsWith("1.7")) {
+      // This test is unreliable on Java 7, probably GC is not triggered by System.gc()
+      return;
+    }
```

Comment:
@himanshug I don't know, but I've seen this many times. Also, reliability of `System.gc()` depends on the platform, fortunately we have only Linux in Druid. But this is not a problem, I've changed the comment to explain why.

Anyway I changed this method to generate some garbage and sleep between calls of `System.gc()` to make it more likely that GC is triggered.